### PR TITLE
add synopsis for 'set license' statement

### PR DESCRIPTION
**Pull request by smakalias**
## Summary of the changes / Why this improves CrateDB

synopsis for 'set license' statement

## Checklist

 - [ ] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: blackbox/docs/sql/statements/alter-cluster.rst [hunk 1]
```diff
@@ -50,3 +51,21 @@ get acknowledged.
     Additionally, keep in mind that this statement only triggers the shard
     re-allocation and is therefore asynchronous. Unassigned shards with large
     size will take some time to allocate.
+
+``SET LICENSE``
+---------------
+
+Registers license information to the cluster state based on the ``license_parameters``::
+
+    {
```
Comment (mfussenegger) [inline blackbox/docs/sql/statements/alter-cluster.rst]: with clauses usually contain assignment pairs separated by comma. `WITH ( expr = val [ , … ] )`

File: blackbox/docs/sql/statements/alter-cluster.rst [hunk 2]
```diff
@@ -50,3 +51,21 @@ get acknowledged.
...
+      expiration_date = value
+      issued_to = value
+      signature = value
+    }
+
+:expiration_date:
+   The date on which the license expires
+:issued_to:
+   The organisation where license is issued
```
Comment (mfussenegger) [inline blackbox/docs/sql/statements/alter-cluster.rst]: `where`? It's the organization for which the license has been issued.

File: blackbox/docs/sql/statements/alter-cluster.rst [hunk 3]
```diff
@@ -50,3 +51,21 @@ get acknowledged.
     Additionally, keep in mind that this statement only triggers the shard
     re-allocation and is therefore asynchronous. Unassigned shards with large
     size will take some time to allocate.
+
+``SET LICENSE``
+---------------
+
+Registers license information to the cluster state based on the ``license_parameters``::
```
Comment (mfussenegger) [inline blackbox/docs/sql/statements/alter-cluster.rst]: `to the cluster state` is imho unnecessary. It's an implementation detail where it is stored or kept. We should focus this in behavior relevant for the user. 

E.g. mention the validation and what happens if it is rejected?

File: blackbox/docs/config/license.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
+.. _conf-license:
+
+================
+License Settings
+================
+
+The section documents settings that only apply to the CrateDB Enterprise License.
+
+``expiration_date``: *expiration_date* (default: ``null``)
```
Comment (mfussenegger) [inline blackbox/docs/config/license.rst]: `default` feels a bit strange - imho all fields should be required. So I'd remove these.

Comment (smakalias) [reply blackbox/docs/config/license.rst]: agree - I'll remove the default

File: blackbox/docs/config/license.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
+.. _conf-license:
+
+================
+License Settings
+================
+
+The section documents settings that only apply to the CrateDB Enterprise License.
```
Comment (andreidan) [inline blackbox/docs/config/license.rst]: "this section"

Comment (smakalias) [reply blackbox/docs/config/license.rst]: copy-paste, the root of all evil. Should I change it in the session doc as well?

File: blackbox/docs/sql/statements/set.rst [hunk 6]
```diff
@@ -47,6 +49,20 @@ logging level.
 ``SET SESSION/LOCAL`` are introduced to be compliant with third-party
 applications which use the PostgresSQL wire protocol.
 
+``SET LICENSE`` registers :ref:`License specific Settings <conf-license>`
+for license verification. If the verification succeeds, the cluster will
+continue to operate in `Enterprise Edition`_ until the license expiry date.
+
+.. NOTE::
+
+    Should the license verification fails, the cluster will continue to
```
Comment (andreidan) [inline blackbox/docs/sql/statements/set.rst]: "fail"

File: blackbox/docs/sql/statements/set.rst [hunk 7]
```diff
@@ -47,6 +49,20 @@ logging level.
...
+    operate in the existing mode. e.g. ``Enterprise Trial`` if the current license
```
Comment (andreidan) [inline blackbox/docs/sql/statements/set.rst]: drop "." after mode as that separates the context of the "next sentence" that is not a sentence on its own

File: blackbox/docs/sql/statements/set.rst [hunk 8]
```diff
@@ -47,6 +49,20 @@ logging level.
...
+    is still valid (i.e. not expired).
+
+    If the verification fails then when the existing licence expires, the cluster
+    will continue to operate, however it will operate in the `Community Edition`_,
```
Comment (andreidan) [inline blackbox/docs/sql/statements/set.rst]: "in Community Edition" or "in the Community Edition _mode_"

**Pull request merged**